warehouse/query-core/src/main/java/datawave/next/ListDocIdIterator.java

```diff
@@ -0,0 +1,177 @@
+ * An alternate implementation of the {@link DatawaveFieldIndexListIteratorJexl}
+ * <p>
+ * Sorted order is not guaranteed.
+ */
+public class ListDocIdIterator extends DocIdIterator {
+
+    private static final Logger log = LoggerFactory.getLogger(ListDocIdIterator.class);
+
+    private List<String> values;
+    private List<Range> ranges;
+    private Range currentRange = null;
+
+    /**
+     * Minimal constructor requires the source, row, and JexlNode
+     *
+     * @param source
+     *            the source iterator
+     * @param row
+     *            the row
+     * @param node
+     *            the JexlNode
+     */
+    public ListDocIdIterator(SortedKeyValueIterator<Key,Value> source, String row, JexlNode node) {
+        super(source, row, node);
+    }
+
+    @Override
+    protected void parseNode(JexlNode node) {
+        QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(node);
+        Preconditions.checkArgument(instance.getType().equals(MarkerType.EXCEEDED_OR));
+
+        ExceededOr exceededOr = new ExceededOr(instance.getSource());
+        this.field = exceededOr.getField();
+        this.values = new ArrayList<>();
+        this.values.addAll(exceededOr.getParams().getValues());
+    }
+
+    private void buildRanges() {
+        this.ranges = new ArrayList<>();
+        Collections.sort(values);
+        for (String value : values) {
+            if (datatypeFilter != null && !datatypeFilter.isEmpty()) {
+                Key start = new Key(this.row, this.prefix, value + "\u0000" + datatypeFilter.first() + "\u0000");
+                Key end = new Key(this.row, this.prefix, value + "\u0000" + datatypeFilter.last() + "\u0000" + Constants.MAX_UNICODE_STRING);
+                ranges.add(new Range(start, true, end, false));
+            } else {
+                Key start = new Key(this.row, this.prefix, value + "\u0000");
+                Key end = new Key(this.row, this.prefix, value + "\u0000" + Constants.MAX_UNICODE_STRING);
+                ranges.add(new Range(start, true, end, false));
+            }
+        }
+        Collections.sort(ranges);
+        Preconditions.checkState(!ranges.isEmpty(), "Range list should not be empty");
+    }
+
+    @Override
+    protected Range createScanRange() {
+        throw new IllegalStateException("ListDocIdIterator does not implement createScanRange()");
+    }
+
+    @Override
+    public boolean hasNext() {
+        if (!seenSeek) {
+            handleFirstSeek();
+        }
+
+        while (tk == null) {
+
+            advanceRangeIfNecessary();
+            if (currentRange == null) {
+                break;
+            }
+
+            Key top = source.getTopKey();
+            parser.parse(top);
+
+            // apply time and datatype filter
+            if (datatypeFilter != null && !datatypeFilter.isEmpty() && !datatypeFilter.contains(parser.getDatatype())) {
+                handleDatatypeMiss();
+                continue;
+            }
```

Comment:
```suggestion
            // apply time and datatype filter
            if (datatypeFilter != null && !datatypeFilter.isEmpty() && !datatypeFilter.contains(parser.getDatatype())) {
                handleDatatypeMiss();
                safeNext();
                continue;
            }
```

Reply:
Part of https://github.com/NationalSecurityAgency/datawave/issues/2916